projects/aca-content/src/lib/components/search/search-input/search-input.component.ts

```diff
@@ -85,26 +86,42 @@ export class SearchInputComponent implements OnInit, OnDestroy {
     private config: AppConfigService,
     private router: Router,
     private store: Store<AppStore>,
-    private appHookService: AppHookService
+    private appHookService: AppHookService,
+    private searchInputService: SearchInputService
   ) {
     this.searchOnChange = this.config.get<boolean>('search.aca:triggeredOnChange', true);
   }
 
   ngOnInit() {
-    this.showInputValue();
-
-    this.router.events
-      .pipe(takeUntil(this.onDestroy$))
-      .pipe(filter((e) => e instanceof RouterEvent))
-      .subscribe((event) => {
-        if (event instanceof NavigationEnd) {
-          this.showInputValue();
-        }
+    if (this.isSearchRoute()) {
+      this.showInputValue();
+
+      this.router.events
+        .pipe(takeUntil(this.onDestroy$))
+        .pipe(filter((e) => e instanceof RouterEvent))
+        .subscribe((event) => {
+          if (event instanceof NavigationEnd) {
+            this.showInputValue();
+          }
+        });
+
+      this.appHookService.library400Error.pipe(takeUntil(this.onDestroy$)).subscribe(() => {
+        this.has400LibraryError = true;
       });
+    }
+  }
+
+  isSearchRoute(): boolean {
+    return this.router.url.includes('/search');
+  }
+
+  navigateToSearch() {
+    this.searchInputService.saveRoute(this.router.url);
+    this.router.navigate(['/search']);
```

Comment:
should probably be handled by the `SearchInputService` if you already started creating some business logic there in the service